modules/dm_Recycler/clients/base/api/RecyclerApi.php

```diff
@@ -71,8 +71,32 @@ public function registerApiRest()
 	 */
 	public function restore($api, $args)
 	{
-		//custom logic
-		return $args;
+
+        if(empty($args['bean_module'])) {
+            throw new SugarApiException("Missing Bean Module Parameter", null, null, 422, 'invalid_parameter');
+        }
+
+        if (empty($args['bean_id'])) {
+            throw new SugarApiException("Missing Bean ID Parameter", null, null, 422, 'invalid_parameter');
+        }
+
+		//Restore recycled record
+        $bean = BeanFactory::newBean($args['bean_module']);
+
+		$beanTableName = $bean->table_name;
+		//TODO: lazy... better way? bean doesnt work, api doesnt work, SugarQuery doesnt do updates yet... also SQL Injection much?
```

Comment:
I'll commit this, and either you can or if I get to it this weekend, look into $db->prepareStatement... prepared statements stop SQL injection (to an extent) so it would probably be the best route for this.